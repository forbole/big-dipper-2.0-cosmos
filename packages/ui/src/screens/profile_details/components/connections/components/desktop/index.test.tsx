import Desktop from '@/screens/profile_details/components/connections/components/desktop';
import MockTheme from '@/tests/mocks/MockTheme';
import { ComponentProps } from 'react';
import renderer from 'react-test-renderer';
import AutoSizer from 'react-virtualized-auto-sizer';

// ==================================
// mocks
// ==================================
jest.mock(
  'react-virtualized-auto-sizer',
  () =>
    ({ children }: ComponentProps<typeof AutoSizer>) =>
      children({
        height: 600,
        width: 600,
      })
);

// ==================================
// unit tests
// ==================================
describe('screen: ProfileDetails/Connections/Desktop', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <MockTheme>
        <Desktop
          items={[
            {
              network: 'native',
              identifier: 'desmos1rzhewpmmdl72lhnxj6zmxr4v94f522s4ff2psv',
              creationTime: '2021-08-31T17:02:28.575104',
            },
            {
              network: 'emoney',
              identifier: 'emoney1wke3ev9ja6rxsngld75r3vppcpet94xxnh63ry',
              creationTime: '2021-08-31T17:02:28.575104',
            },
          ]}
        />
      </MockTheme>
    );
    const tree = component?.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
