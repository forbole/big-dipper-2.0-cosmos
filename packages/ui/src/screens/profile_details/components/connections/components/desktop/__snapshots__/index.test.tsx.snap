// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: ProfileDetails/Connections/Desktop matches snapshot 1`] = `
.emotion-0 {
  display: table;
  width: 100%;
  border-collapse: collapse;
  border-spacing: 0;
}

.emotion-0 caption {
  font-size: 0.875rem;
  letter-spacing: 0.25px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  line-height: 1.43;
  padding: 16px;
  color: #414141;
  text-align: left;
  caption-side: bottom;
}

.emotion-1 {
  display: table-header-group;
  background-color: initial;
}

.emotion-2 {
  color: inherit;
  display: table-row;
  vertical-align: middle;
  outline: 0;
}

.emotion-2.MuiTableRow-hover:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

.emotion-2.Mui-selected {
  background-color: rgba(255, 131, 91, 0.08);
}

.emotion-2.Mui-selected:hover {
  background-color: rgba(255, 131, 91, 0.12);
}

.emotion-3 {
  font-size: 0.875rem;
  letter-spacing: 0.25px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  line-height: 1.5rem;
  display: table-cell;
  vertical-align: inherit;
  border-bottom: 1px solid rgba(252, 252, 252, 1);
  text-align: left;
  padding: 16px;
  color: #000000;
  border-bottom: none;
  padding: 0 16px;
  height: 50px;
  font-size: 1rem;
}

.emotion-5 {
  font-size: 0.875rem;
  letter-spacing: 0.25px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  line-height: 1.5rem;
  display: table-cell;
  vertical-align: inherit;
  border-bottom: 1px solid rgba(252, 252, 252, 1);
  text-align: right;
  padding: 16px;
  color: #000000;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  border-bottom: none;
  padding: 0 16px;
  height: 50px;
  font-size: 1rem;
}

.emotion-6 {
  display: table-row-group;
}

.emotion-6 .MuiTableRow-root:nth-of-type(odd) {
  background-color: #F8F8F8;
}

.emotion-6 .MuiTableCell-root {
  color: #414141;
}

.emotion-8 {
  font-size: 0.875rem;
  letter-spacing: 0.25px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  line-height: 1.43;
  display: table-cell;
  vertical-align: inherit;
  border-bottom: 1px solid rgba(252, 252, 252, 1);
  text-align: left;
  padding: 16px;
  color: #000000;
  border-bottom: none;
  padding: 0 16px;
  height: 50px;
  font-size: 1rem;
}

.emotion-10 {
  font-size: 0.875rem;
  letter-spacing: 0.25px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  line-height: 1.43;
  display: table-cell;
  vertical-align: inherit;
  border-bottom: 1px solid rgba(252, 252, 252, 1);
  text-align: right;
  padding: 16px;
  color: #000000;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  border-bottom: none;
  padding: 0 16px;
  height: 50px;
  font-size: 1rem;
}

<div>
  <table
    className="MuiTable-root emotion-0"
    role={null}
  >
    <thead
      className="MuiTableHead-root emotion-1"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head emotion-2"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium emotion-3"
          scope="col"
          style={
            {
              "width": "25%",
            }
          }
        >
          network
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium emotion-3"
          scope="col"
          style={
            {
              "width": "50%",
            }
          }
        >
          identifier
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium emotion-5"
          scope="col"
          style={
            {
              "width": "25%",
            }
          }
        >
          creationTime
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root emotion-6"
      role={null}
    >
      <tr
        className="MuiTableRow-root emotion-2"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium emotion-8"
          style={
            {
              "width": "25%",
            }
          }
        >
          NATIVE
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium emotion-8"
          style={
            {
              "width": "50%",
            }
          }
        >
          <a
            className="value"
            href="/accounts/desmos1rzhewpmmdl72lhnxj6zmxr4v94f522s4ff2psv"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            desmos1rzhewpmmdl72lhnxj6zmxr4v94f522s4ff2psv
          </a>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium emotion-10"
          style={
            {
              "width": "25%",
            }
          }
        >
          2020-08-10 12:00:00
        </td>
      </tr>
      <tr
        className="MuiTableRow-root emotion-2"
        role={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium emotion-8"
          style={
            {
              "width": "25%",
            }
          }
        >
          EMONEY
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium emotion-8"
          style={
            {
              "width": "50%",
            }
          }
        >
          emoney1wke3ev9ja6rxsngld75r3vppcpet94xxnh63ry
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium emotion-10"
          style={
            {
              "width": "25%",
            }
          }
        >
          2020-08-10 12:00:00
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
